BorrowBookUI.java

```diff
@@ -2,116 +2,116 @@
-				}
-				catch (NumberFormatException e) {
-					output("Invalid Member Id");
-				}
-				break;
-
-				
-			case RESTRICTED:
-				input("Press <any key> to cancel");
-				CONTROL.cancel();
-				break;
-			
-				
-			case SCANNING:
-				String Book_Str = input("Scan Book (<enter> completes): ");
-				if (Book_Str.length() == 0) {
-					CONTROL.Complete();
-					break;
-				}
-				try {
-					int BiD = Integer.valueOf(Book_Str).intValue();
-					CONTROL.Scanned(BiD);
-					
-				} catch (NumberFormatException e) {
-					output("Invalid Book Id");
-				} 
-				break;
-					
-				
-			case FINALISING:
-				String Ans = input("Commit loans? (Y/N): ");
-				if (Ans.toUpperCase().equals("N")) {
-					CONTROL.cancel();
-					
-				} else {
-					CONTROL.Commit_LOans();
-					input("Press <any key> to complete ");
-				}
-				break;
-				
-				
-			case COMPLETED:
-				output("Borrowing Completed");
-				return;
-	
-				
-			default:
-				output("Unhandled state");
-				throw new RuntimeException("BorrowBookUI : unhandled state :" + StaTe);			
-			}
-		}		
-	}
-
-
-	public void Display(Object object) {
-		output(object);		
-	}
+
+    public static enum UIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+    private BorrowBookControl control;
+    private Scanner input;
+    private UIState state;
+
+
+    public BorrowBookUI(BorrowBookControl control) {
+        this.control = control;
+        this.input = new Scanner(System.in);
+        this.state = UIState.INITIALISED;
+        this.control.setUI(this);
+    }
+
+
+    private String input(String prompt) {
+        System.out.print(prompt);
+        return this.input.nextLine();
+    }
+
+
+    private void output(Object object) {
+        System.out.println(object);
+    }
+
+
+    public void setState(UIState state) {
+        this.state = state;
+    }
+
+
+    public void run() {
+        this.output("Borrow Book Use Case UI\n");
+
+        while (true) {
+
+            switch (this.state) {
+
+            case CANCELLED:
+                this.output("Borrowing Cancelled");
+                return;
+
+
+            case READY:
+                String memberCard = input("Swipe member card (press <enter> to cancel): ");
+                if (memberCard.length() == 0) {
+                    this.control.cancel();
+                    break;
+                }
+                try {
+                    int memberId = Integer.valueOf(memberCard).intValue();
+                    this.control.swiped(memberId);
+                }
+                catch (NumberFormatException e) {
+                    this.output("Invalid Member Id");
+                }
+                break;
+
+
+            case RESTRICTED:
+                this.input("Press <any key> to cancel");
+                this.control.cancel();
+                break;
+
+
+            case SCANNING:
+                String scanBook = input("Scan Book (<enter> completes): ");

```

Comment:
D11: ensure the variable name is a NOUN 
e.g. 'scannedBook'

Reply:
updated as per example